dsc-linux-5.4.173/patches-2023-02/00_penfw.patch

```diff
@@ -0,0 +1,795 @@
+From 81bcc0f4ce6f562f3e161700b4c731094a8739d5 Mon Sep 17 00:00:00 2001
+From: Hiren Mehta <hiren@pensando.io>
+Date: Wed, 15 Feb 2023 18:04:24 -0800
+Subject: [PATCH] drivers/soc/pensando: penfw driver
+
+penfw driver provides sysfs and ioctl interface for the userspace
+to communicate to the bl31 secure monitor
+
+Signed-off-by: Hiren Mehta <hiren@pensando.io>
+---
+ arch/arm64/boot/dts/pensando/elba-asic.dts |   8 +
+ arch/arm64/configs/elba_defconfig          |   2 +-
+ drivers/soc/pensando/Kconfig               |   6 +
+ drivers/soc/pensando/Makefile              |   1 +
+ drivers/soc/pensando/penfw.h               |  35 ++
+ drivers/soc/pensando/penfw_drv.c           | 145 +++++++
+ drivers/soc/pensando/penfw_smc.c           |  63 ++++
+ drivers/soc/pensando/penfw_sysfs.c         | 420 +++++++++++++++++++++
+ drivers/soc/pensando/penfw_sysfs.h         |  11 +
+ 9 files changed, 690 insertions(+), 1 deletion(-)
+ create mode 100644 drivers/soc/pensando/penfw.h
+ create mode 100644 drivers/soc/pensando/penfw_drv.c
+ create mode 100644 drivers/soc/pensando/penfw_smc.c
+ create mode 100644 drivers/soc/pensando/penfw_sysfs.c
+ create mode 100644 drivers/soc/pensando/penfw_sysfs.h
+
+diff --git a/arch/arm64/boot/dts/pensando/elba-asic.dts b/arch/arm64/boot/dts/pensando/elba-asic.dts
+index 70d0dbd3887b..a1ba7b245949 100644
+--- a/arch/arm64/boot/dts/pensando/elba-asic.dts
++++ b/arch/arm64/boot/dts/pensando/elba-asic.dts
+@@ -24,3 +24,11 @@
+ #include "elba-flash-parts.dtsi"
+ #include "elba-asic-memory.dtsi"
+ #include "elba-mdev.dtsi"
++
++/ {
++       firmware {
++               penfw {
++                       compatible = "pensando,penfw";
++               };
++       };
++};
+diff --git a/arch/arm64/configs/elba_defconfig b/arch/arm64/configs/elba_defconfig
+index 5e26a8129a46..f9f051a7a241 100644
+--- a/arch/arm64/configs/elba_defconfig
++++ b/arch/arm64/configs/elba_defconfig
+@@ -229,7 +229,6 @@ CONFIG_WATCHDOG=y
+ CONFIG_SOFT_WATCHDOG=y
+ CONFIG_DW_WATCHDOG=y
+ CONFIG_MFD_PENSANDO_ELBASR=y
+-CONFIG_MFD_SYSCON=y
```

Comment:
You decided to include this cleanup in this patch?  Intentional?  I could go either way, but I wasn't expecting to see this after our chat.

Reply:
I could either way as well. But I thought, it will keep future patches clean.

Reply:
You could structure this as 2 patches, 1) 00_penfw and 2) 01_elba_defconfig_cleanup so it is clear what is penfw, but up to you.